pkg/runtime/runtime-manifest.go

```diff
@@ -403,35 +430,47 @@ func (m *RuntimeManifest) buildSourceData() (map[string]interface{}, error) {
 
 // ResolveStep will walk through the Step's data and resolve any placeholder
 // data using the definitions in the manifest, like parameters or credentials.
-func (m *RuntimeManifest) ResolveStep(step *manifest.Step) error {
-	mustache.AllowMissingVariables = false
+func (m *RuntimeManifest) ResolveStep(stepIndex int, step *manifest.Step) error {
+	if err := m.loadManifest(); err != nil {
+		return err
+	}
+
+	// Refresh our template data
 	sourceData, err := m.buildSourceData()
 	if err != nil {
 		return fmt.Errorf("unable to build step template data: %w", err)
 	}
 
-	if m.Debug {
-		fmt.Fprintf(m.Err, "=== Step Data ===\n%v\n", sourceData)
-	}
-
-	payload, err := yaml.Marshal(step)
+	// Get the original yaml for the current step
+	stepPath := fmt.Sprintf("%s[%d]", m.Action, stepIndex)
+	stepNode, err := m.manifestYAML.GetNode(stepPath)
 	if err != nil {
-		return fmt.Errorf("invalid step data %v: %w", step, err)
+		return fmt.Errorf("unable to retrieve original yaml for step %s: %w", stepPath, err)
+	}
+	var stepTemplate bytes.Buffer
+	enc := yaml3.NewEncoder(&stepTemplate)
+	defer enc.Close()
+	if err := enc.Encode(stepNode); err != nil {
+		return fmt.Errorf("error re-encoding porter.yaml for templating: %w", err)
 	}
 
 	if m.Debug {
-		fmt.Fprintf(m.Err, "=== Step Template ===\n%v\n", string(payload))
+		fmt.Fprintf(m.Err, "=== Step Data ===\n%v\n", sourceData)
```

Comment:
why do we not using the logger here?

Reply:
I decided against redoing the context plumbing here as it would have exploded out this refactoring PR even further. I can address that in a separate PR.